src/loaddb/load_db.c

```diff
@@ -1288,11 +1306,26 @@ load_object_file (load_args * args)
     }
 
   /* *INDENT-OFF* */
-  batch_handler b_handler = [] (const batch &batch) -> int
+  batch_handler b_handler = [&] (const batch &batch) -> int
   {
-    int ret = loaddb_load_batch (batch);
-    delete &batch;
+    int ret = loaddb_load_batch (batch, false);
+    bool not_batch_accepted = ret == ER_LDR_INVALID_STATE;
```

Comment:
it is preferable to use a flag instead of error code to pass batch accepted/not accepted information.